### ISSUE TITLE
Does the game work for 2 players?

### ISSUE THREAD

**Issue by erbridge** [to-the-end/to-the-end#87]
Would it be better off with one player and the game getting in the way in a semi-predictable way? For example, using the chain if you bump into a wall, or raising bits of floor more often as you get close to the torches (like it's resisting you).

**Comment by Archeks** [comment]
I think, what matters is, the game should be nice to play.
in my opinion: 
for a single player - it should have really nice graphics, and all kind of PVE staff
for multiplayer - graphics level doesn't matter as much as game mechanics.

and multiplayer is more fun to play usually

**Comment by erbridge** [comment]
For me there is more to this game than just the mechanics. The narrative is hugely important for the experience, I think. If we'd built an arcade / arena / physics / heavily mechanical game, then I'd say your estimation is right, but since we need to balance both things, it would be wrong to focus entirely on the mechanics. And given the narrative focus, we need to make sure the game keeps the mood created by the cutscenes through the gameplay, which implies to me that it might be better as a single player experience.

Also, I think it's important to be aware that fun doesn't have to come from the game mechanics. Interactive novels usually have next to no mechanics, but are still compelling and entertaining. This game has much more interactivity than that, but the mechanics should support the narrative to make a compelling experience that is deeper than either one in isolation.

Having said that, it might be possible to make it work as a multiplayer game, but I just wanted to open the discussion about whether that is the right thing to be doing.

**Comment by daveallenbpm** [comment]
With what we currently have single player makes more sense to me.

We have something quite atmospheric and primarily story driven. I can't help but feel the multiplayer aspect detracts from the story and atmosphere.

The difficulty in multiplayer is that there's two separate experiences that need to be equally engaging, whereas with single player we can focus on making an immersive experience.

Not that I'm altogether against having a multiplayer! I'd love to have a multiplayer experience with this,  but developing the single player makes more sense to me initially - that way we can develop and understand the core mechanics, and then use this knowledge to create a richer multiplayer experience.

**Comment by duncjam** [comment]
Yeah I'm inclined to agree with @erbridge and @daveallenbpm wrt this. 

I think the multiplayer is a really interesting idea in terms of having a second force restricting the movement of the protagonist (both in terms of drawing a part of the narrative thread into the mechanics [conflict etc.], and as a way of making the play more varied) but I think it shatters the illusion a little bit while playing. E.g. I've shown it to my gf and a few friends and I demoed the second player mechanics to them (which they thought were cool!), but then just ended up leaving them to it because they wanted to experience the narrative, and because in all honesty, the fun wears off for the second player when the initial novelty does, I think. 

So I'd suggest potentially we try our hand at a single player version - we could always (here comes Captain Doesn'tunderstandthetechnicalities) try separate builds or modes? i.e. remodel it as a one player experience, and once we're happy with that, create a separate version with the 2nd player controls dropped back in? Either as a sort of Arcade Mode version, or just a separate game?

I like the idea of the environment still fighting you though, and I think having it done by the game procedurally is a really good idea (we'd just have to be careful in terms of weighting so it doesn't get too frustrating). 

In terms of ways to keep/make it more engaging as a single player experience, here's a few ideas on top of the ones already built (open to criticism, revision, rejection):

 - Escalation: As said above, it could potentially develop as it goes along. This makes sense level by level, but also - I think - by the two halves of the game

1st half:  I think given the music and the narrative elements of the first couple of levels, we're probably looking at something more dreamlike and slow paced there, so it wouldn't really make sense to have a very aggressive environment at that stage - some hindrances but have it more maze based (conceptually, both in terms of narrative and player experience, we're still discovering rather than fearing here - I think this at least holds for the 1st level, if not the 1st half)

2nd half: As the music gets more foreboding I think it then makes sense to then ramp up the interruptions, difficulty, general sense of frenetics/panic (3 to 4th level transition, might it be possible to add more layers to the music, get it really thrumming?)

 - Potential mechanics:

Tighter maze: As @erbridge has said, I think a tighter maze would make sense, and would allow for the player to experience a greater sense of risk if there were some environmental threats (moving towards the PVE idea @Archeks suggested)?

Contact with walls - could either attach the chain or cause damage or both (if we decided to replace time with damage, or introduce it separately - not sure how i feel about that but just to put it out there)

"Sinkholes" - not something you can disappear into, but a gravitational sort of vector around a point at the centre of a circle, which would make it more difficult, pull you closer to walls etc. Don't have to be drastic, but something to add another factor to play (they use them a lot in A Link to the Past, I seem to remember?)

'Simon Says' Blocks/Opacities - I described this (sort of) at the first brainstorm, but either a block could appear in front of the player (or a prompt which greys out the rest of the screen, but that might ruin immersion) which shows a pattern on it's top, and then the player has to enter the same pattern back using WASD or other keys

I had another one, but I've forgotten it. Anyway, I need to get back to my actual job now haha, but just a few thoughts. Interested to hear thoughts.

**Comment by duncjam** [comment]
Jesus that was a long message

**Comment by erbridge** [comment]
My first thought was "holy shit!" :).

I think those suggestions are really good. I hadn't thought of it like that before, but making the first level a sort of cross between an introduction to the narrative and a tutorial to teach the player the goal without the obstacles is a really nice idea. That sits really well with me.

In terms of progression, having one new mechanic on each level would probably work well, with some levels perhaps combining mechanics from previous levels (I think @duncjam suggested he had come up with up to 10 levels of content if we needed it). Levels can be longer or shorter than each other to vary the pacing, too.

I think keeping away from explicit damage would be best, but we could use the timer as a sort of health bar instead, so "damage" is a reduction in the time limit, or something like that.

Another idea for an obstacle is walls that flicker in and out of existence, so you have to time your movement to get through them.

Nice ideas, @duncjam!

**Comment by erbridge** [comment]
And the chain could be based on distance, so it slows you down until you "break" it by  reaching its threshold (with a bit of resistance at that point).

**Comment by erbridge** [comment]
Oh, and in terms of the multiplayer mode, we could do a separate "arena" mode that allows for a player (or many players if we hooked into online modes?) to control the environment in a more fast-paced party game style. That could either be to play through the story, or to just solve puzzles in isolation.

Anyway. Nice ideas! I'd like to see what everyone else thinks, though. @endritbajo @mhle @2easy4sam @kanianto?

**Comment by kanianto** [comment]
Hello,

I believe we should work on a one player campaign in the beginning with the
choice of a second player as "a plus".

Even so I quite like the main concept of a second player (or more players)
which use the environment against the first player; it's actually the
coolest thing.
We still have to think that the game should be available to play for just
one player as well so I think an AI for the second player is mandatory.

The arena idea it's quite cool but I think we should have a focus on the
campaign first.  Once we set priorities and we achieve them it'll be easier
to decide for cool extra :).

The last thing we want is to loose the main track

**Comment by Archeks** [comment]
Hi guys,
Maybe you are right, we should focus on PVE first.
Ill try to find some time to get into it on weekends.
I've got some ideas for a single player.

1. Items, when you pick them up, you get some buffs. (More time
(effectively it will be similar to health), faster movement speed, time
shift)
2. Teleport doors, when you enter it you appear somewhere else. (can be
random aswell)
3. Maybe we need to add character selection at the beginning (so we have a
girl character and a boy, or several of each with different "skills" ),

**Comment by erbridge** [comment]
Those are cool ideas @Archeks, but let's make sure they fit into the context of the narrative.

Re point 3: let's make sure the characters don't have abilities based on their genders... I'm actually more inclined to make at least the main / base character androgynous, so their gender is undefined, though that relies on @kanianto to make us a cool sprite :).

**Comment by Archeks** [comment]
For the single player (pve), maybe we need to add some kind of boss Fights

**Comment by erbridge** [comment]
I'm not sure boss fights would work, at least in the traditional sense. The idea is that the player is solving puzzles while the environment tries to stop them, representing an internal struggle. I think presenting the player with an enemy would undermine that.
